### ISSUE TITLE
Support Code-Coverage (preferably Jacoco) for integration tests

### ISSUE THREAD

**Issue by tonit** [bndtools/bndtools#1434]
Integration tests in bndtools use a custom tets launcher. Thats why they are not part of test coverage (coverage always 0%). 
Guess its only a matter of passing jacoco preprocessing code to the underlying test process. I have not looked deeply into the launcher code. Maybe there is already a way to do it but i do not find the right documentation for it?

**Comment by bjhargrave** [comment]
Please make a PR, thanks.

**Comment by bjhargrave** [comment]
Also, have you tried the jacoco gradle plugin in your gradle build?

**Comment by tonit** [comment]
Yes we did. Integrationtests do not contribute to the code coverage measured by jacoco.
Detailed description: We have regular unit tests and integration tests. The Jacoco Report shows unit tests as well as integration tests. But only the unit tests contribute to the coverage. Both test types are executed.
Without having looked at the code yet, i guess integration tests run in their own jvm?

**Comment by bjhargrave** [comment]
> i guess integration tests run in their own jvm?

Yes, a new VM is launched to run the framework, test bundles and any other -runbundles.

**Comment by jwausle** [comment]
It seems that `-javaagent: jacoco.jar...` instruction is wrong evaluated while osgi test execution.
- the calculated java cmd looks like `java -javaagent:biz.aQute.launcher-3.4.0.jar -cp ...:biz.aQute.launcher-3.4.0.jar ...` 
- the ./bnd.bnd "-javaagent:..." instruction value is missed

I prepared a sample project

```
git clone https://github.com/jwausle/bndtools-testcase-fragment.git --jacoco-support
cd bndtools-testcase-fragment
sh gradlew clean build
```

**Comment by bjhargrave** [comment]
You are using the `-javaagent` Bnd instruction incorrectly. The documentation was wrong and I have made a PR to fix it. 

You need to put `jacoco.jar` (if that is the agent jar) on the `-runpath` and then have`-javaagent: true`.  Then the launcher will inspect the classpath elements create `-javaagent:` Java arguments for all the jars having a `Premain-Class` manifest header.
